### DISCUSSION TITLE
Custom Data Provider (Budibase plugin)

### DISCUSSION THREAD

**Discussion by helicop0420** [Budibase/budibase#10648] [Help]
Hello Everyone
I made custom dataprovider.
but that's children(for ex. data repeater) can't get data from this custom plugin.
![image](https://github.com/Budibase/budibase/assets/116023731/e24dcd7c-8a60-4b45-9d57-9241de420529)

I think it's related to schema.json in plugin development.
hope your kind help.
Thanks.

**Comment by melohagan** [comment]
Hey @helicop0420 

Are you providing the context in your schema? https://docs.budibase.com/docs/custom-component#example-context-provider

For example the Budibase Dataprovider looks like this:

```json
    "hasChildren": true,
    "actions": ["RefreshDatasource"],
    "context": {
      "type": "static",
      "values": [
        {
          "label": "Rows",
          "key": "rows",
          "type": "array"
        },
        {
          "label": "Extra Info",
          "key": "info",
          "type": "string"
        },
        {
          "label": "Rows Length",
          "key": "rowsLength",
          "type": "number"
        },
        {
          "label": "Schema",
          "key": "schema",
          "type": "object"
        },
        {
          "label": "Page Number",
          "key": "pageNumber",
          "type": "number"
        }
      ]
    }
```

I also noticed this line of code in the `DataProviderSelect`:

```js
$: providers = path.filter(c => c._component?.endsWith("/dataprovider"))
```

So you may need to name your plugin such that it ends with `"/dataprovider"`.

**Reply by helicop0420** [reply]
Thanks for your consideration. @melohagan 
One question. How to name my plugin with "/dataprovider"?
Should I change schema.json?
This is my current schema.json.
{
  "type": "component",
  "metadata": {},
  "schema": {
    "name": "Infinite-dataprovider",
    "icon": "Data",
    "illegalChildren": ["section"],
    "hasChildren": true,
    "actions": ["RefreshDatasource"],
...

**Reply by melohagan** [reply]
Hey @helicop0420 

It would be the *name* prop. So `"name": "infinite/dataprovider"`

I'm not sure if this will work, but could be worth a try!
